src/app/pages/ArticlePage/experimentTopStories/helpers.tsx

```diff
@@ -66,23 +68,52 @@ const enableExperimentTopStories = ({
   );
 };
 
+const insertBlockAtPosition = (
+  blocks: OptimoBlock[],
+  blockToInsert: OptimoBlock,
+  position: 'Quarter' | 'Half' | 'ThreeQuarters',
+) => {
+  const insertionPercentages = {
+    Quarter: 0.25,
+    Half: 0.5,
+    ThreeQuarters: 0.75,
+  };
+  const percentage = insertionPercentages[position];
+
+  const calculatedIndex = Math.floor((blocks.length - 1) * percentage); // -1 accounts for 'wsoj' block which is never rendered on PS articles
+  const insertIndex = Math.max(calculatedIndex, 3); // Ensure insertIndex is at least 3
```

Comment:
**Suggestion (non-blocking)** You could remove these comments if you extracted the blocks.length - 1 and 3 into variables with explicit names.

The second comment is not adding much either way as it's just describing what the code does!